src/game/Unit.cpp

```diff
@@ -9596,7 +9596,7 @@ void Unit::SetIncapacitatedState(bool apply, uint32 state, ObjectGuid casterGuid
 
     Player* controller = GetCharmerOrOwnerPlayerOrPlayerItself();
     const bool control = controller ? controller->IsClientControl(this) : false;
-    const bool movement = (state != UNIT_FLAG_STUNNED);
+    const bool movement = (state != UNIT_FLAG_STUNNED) || !apply;
```

Comment:
At first view it seem correct but i will delegate that change to @Warlockbugs because i know first view is 90% of the time not enough about that portion of the code.